runtime/src/tiered_storage/writer.rs

```diff
@@ -0,0 +1,76 @@
+//! docs/src/proposals/append-vec-storage.md
+
+use {
+    crate::{
+        account_storage::meta::{StorableAccountsWithHashesAndWriteVersions, StoredAccountInfo},
+        storable_accounts::StorableAccounts,
+        tiered_storage::{
+            file::TieredStorageFile,
+            footer::{AccountMetaFormat, TieredStorageFooter},
+            hot::HotAccountMeta,
+            meta::TieredAccountMeta,
+            TieredStorageFormat,
+        },
+    },
+    solana_sdk::{account::ReadableAccount, hash::Hash},
+    std::{borrow::Borrow, fs::remove_file, path::Path},
+};
+
+#[derive(Debug)]
+pub struct TieredStorageWriter<'format> {
+    storage: TieredStorageFile,
+    format: &'format TieredStorageFormat,
+}
+
+impl<'format> TieredStorageWriter<'format> {
+    pub fn new(file_path: &Path, format: &'format TieredStorageFormat) -> Self {
+        let _ignored = remove_file(file_path);
+        Self {
+            storage: TieredStorageFile::new_writable(file_path),
+            format,
+        }
+    }
+
+    fn append_accounts_impl<
+        'a,
+        'b,
+        T: ReadableAccount + Sync,
+        U: StorableAccounts<'a, T>,
+        V: Borrow<Hash>,
+        W: TieredAccountMeta,
+    >(
+        &self,
+        _accounts: &StorableAccountsWithHashesAndWriteVersions<'a, 'b, T, U, V>,
+        footer: TieredStorageFooter,
+        mut _account_metas: Vec<W>,
+        _skip: usize,
+    ) -> Option<Vec<StoredAccountInfo>> {
+        footer.write_footer_block(&self.storage).ok()?;
+        None
+    }
+
+    pub fn append_accounts<
```

Comment:
We don't.  This can be called only once.   The call-once logic is added in [here](https://github.com/solana-labs/solana/pull/32579#discussion_r1274081285) in the next PR.